kie-server-parent/kie-server-services/kie-server-services-jbpm-queries/src/main/java/org/kie/server/services/jbpm/queries/util/QueryStrategy.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm.queries.util;
+
+import java.util.Map;
+
+import org.jbpm.services.api.query.model.QueryParam;
+
+/**
+ * Provides DB specific query and column-mappings for queries.
+ * 
+ * @author <a href="mailto:duncan.doyle@redhat.com">Duncan Doyle</a>
+ */
+public interface QueryStrategy {
```

Comment:
Basically the API uses this concept: http://mswiderski.blogspot.nl/2016/01/advanced-queries-in-kie-server.html

This allowed us to reuse all of the existing API constructs to implement those filters.